processor/groupbytraceprocessor/config.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbytraceprocessor
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// Config is the configuration for the processor.
+type Config struct {
+	configmodels.ProcessorSettings `mapstructure:",squash"`
+
+	// NumTraces is the max number of traces to keep in memory waiting for the duration.
+	// Default: 1.000.000.
+	NumTraces int `mapstructure:"num_traces"`
+
+	// WaitDuration tells the processor to wait for the specified duration for the trace to be complete.
+	// Default: 1s.
+	WaitDuration time.Duration `mapstructure:"wait_duration"`
+
+	// DiscardOrphans instructs the processor to discard traces without the root span.
+	// This typically indicates that the trace is incomplete.
+	// Default: false.
+	// Not yet implemented, and an error will be returned when this option is used.
+	DiscardOrphans bool `mapstructure:"discard_orphans"`
+
+	// StoreInDisk tells the processor to keep only the trace ID in memory, serializing the trace spans to disk.
+	// Useful when the duration to wait for traces to complete is high.
+	// Default: false.
+	// Not yet implemented, and an error will be returned when this option is used.
```

Comment:
If it's not implemented, why do we need to add it right now? Better to add it in the PR adding support for storing on disk? The implementation of that doesn't seem trivial, so I would just remove this parameter for now.

Reply:
I'll remove this before the final review.